framework/src/actions/ProjectedStatefulMaterialStorageAction.C

```diff
@@ -0,0 +1,86 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "ProjectedStatefulMaterialStorageAction.h"
+#include "RankFourTensorForward.h"
+#include "RankTwoTensorForward.h"
+#include "AddAuxVariableAction.h"
+#include "AddVariableAction.h"
+#include "FEProblemBase.h"
+#include "Factory.h"
+
+#include "libmesh/string_to_enum.h"
+
+// we need to add these variables after the material property types are known
+registerMooseAction("MooseApp",
+                    ProjectedStatefulMaterialStorageAction,
+                    "setup_projected_properties");
+
+registerMooseAction("MooseApp", ProjectedStatefulMaterialStorageAction, "add_aux_kernel");
+
+InputParameters
+ProjectedStatefulMaterialStorageAction::validParams()
+{
+  InputParameters params = Action::validParams();
+
+  params.addClassDescription("Mark material properties for projected stateful storage.");
+
+  params.addParam<MooseEnum>(
+      "family",
+      MooseEnum("LAGRANGE MONOMIAL", "LAGRANGE"),
```

Comment:
Let's at least add `L2_LAGRANGE` and `L2_HIERARCHIC` as other options

Reply:
are both of those elemental?

Reply:
they are both discontinuous yes